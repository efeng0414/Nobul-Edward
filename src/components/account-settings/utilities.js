import React from "react";
import SettingsFieldItem from "../settings/common/settings-field-item";
import {
  FIRST_NAME,
  LAST_NAME,
  PHONE,
  ADDRESS1,
  ADDRESS2,
  COUNTRY,
  PROVINCE_OR_STATE,
  CITY,
  POSTAL_OR_ZIP_CODE,
  PASSWORD,
  EMAIL
} from "../../../core/api-transform/users";
import TextField from "../form-fields/text-field";
import PhoneField from "../form-fields/phone-field";
import PostalField from "../form-fields/postal-field";
import CountryField from "../form-fields/country-field";
import ProvinceField from "../form-fields/province-field";
import ChangePasswordForm from "../forms/common/change-password-form";
import ChangeEmailForm from "../forms/common/change-email-form";

const LOCALE_KEY_PREFIX = "consumerSettings.account.";

export const fieldsMap = {
  [FIRST_NAME]: {
    labelKey: "firstName",
    type: TextField,
    required: true
  },
  [LAST_NAME]: {
    labelKey: "lastName",
    type: TextField,
    required: true
  },
  [EMAIL]: {
    labelKey: "email",
    type: ChangeEmailForm,
    required: true
  },
  [PASSWORD]: {
    labelKey: "password",
    type: ChangePasswordForm,
    required: true
  },
  [PHONE]: {
    labelKey: "phoneNumber",
    type: PhoneField,
    required: true
  },
  [ADDRESS1]: {
    labelKey: "address1",
    type: TextField,
    halfWidth: true,
    required: true
  },
  [ADDRESS2]: {
    labelKey: "address2",
    type: TextField,
    halfWidth: true,
    required: false
  },
  [COUNTRY]: {
    labelKey: "country",
    type: CountryField,
    halfWidth: true,
    required: true
  },
  [PROVINCE_OR_STATE]: {
    labelKey: "province",
    type: ProvinceField,
    halfWidth: true,
    required: true
  },
  [CITY]: {
    labelKey: "city",
    type: TextField,
    halfWidth: true,
    required: true
  },
  [POSTAL_OR_ZIP_CODE]: {
    labelKey: "postcode",
    type: PostalField,
    halfWidth: true,
    required: true
  }
};

export const generateSettingFieldsItemForProfile = ({
  userProfile,
  onEditClick
}) => {
  return Object.entries(fieldsMap).map(([key, { labelKey, halfWidth }]) => {
    return (
      <SettingsFieldItem
        labelKey={`${LOCALE_KEY_PREFIX}${labelKey}`}
        value={userProfile[key]}
        onEditClick={onEditClick}
        key={key}
        id={key}
        halfWidth={halfWidth}
      />
    );
  });
};
