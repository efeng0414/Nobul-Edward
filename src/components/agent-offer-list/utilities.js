import { REBATE_COMMISSION } from "../../../core/api-transform/users";
import { REBATE_COMMISSION as OFFER_REBATE_COMMISSION } from "../../../core/api-transform/offers";
import { BUY, SELL } from "../../../core/api-transform/jobs";

export const getJobType = ({ jobDetails, jobTypeProp }) => {
  if (jobTypeProp) return jobTypeProp;
  return jobDetails[REBATE_COMMISSION] !== undefined ||
    jobDetails[OFFER_REBATE_COMMISSION] !== undefined
    ? BUY
    : SELL;
};
