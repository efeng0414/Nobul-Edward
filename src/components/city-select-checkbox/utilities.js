// Helper function to toggle on and off all neighborhoods of the polygon data.
// This is only used to turn on/off Toronto polygons at the moment.
export const toggleSelectedNeighborhoodsHelper = ({
  selectAll,
  mapPolygons,
  currentPolygons,
  setPolygons
}) => {
  const neighbourhoods = mapPolygons ? mapPolygons.neighbourhoods.features : [];

  if (selectAll) {
    const selectedPolygons = {};

    neighbourhoods.map(
      polygon =>
        (selectedPolygons[polygon.properties.polygonId] = {
          name: polygon.properties.name,
          provinceOrState: polygon.properties.provinceOrState,
          country: polygon.properties.country
        })
    );

    setPolygons({
      ...currentPolygons,
      ...selectedPolygons
    });
  } else {
    const selectedPolygons = { ...currentPolygons };
    neighbourhoods.map(
      polygon => delete selectedPolygons[polygon.properties.polygonId]
    );
    setPolygons(selectedPolygons);
  }
};

const neighborhoodsPolygonsMap = new WeakMap();

export const allNeighborhoodsSelected = ({ mapPolygons, currentPolygons }) => {
  if (!mapPolygons || !mapPolygons.neighbourhoods) {
    return false; // No neighborhoods to be selected.
  }

  let neighborhoodPolygonIds = mapPolygons
    ? neighborhoodsPolygonsMap.get(mapPolygons)
    : null;

  if (!neighborhoodPolygonIds) {
    const neighbourhoods = mapPolygons.neighbourhoods.features;
    neighborhoodPolygonIds = neighbourhoods.map(
      polygon => polygon.properties.polygonId
    );
    neighborhoodsPolygonsMap.set(mapPolygons);
  }

  const selectedPolygonIds = Object.keys(currentPolygons);
  if (selectedPolygonIds.length < neighborhoodPolygonIds.length) {
    return false; // If there are less polygons selected than neighborhoods, they sure arent all selected.
  }

  // Check if all neighborhood IDs are included in selectedPolygonIds
  return neighborhoodPolygonIds.every(
    elem => selectedPolygonIds.indexOf(elem) > -1
  );
};

// Check for Toronto, show/hide city button. Will change to generic in future.
export const showCityCheckbox = ({ placesApiResult }) => {
  return (
    placesApiResult.address_components &&
    placesApiResult.address_components[0].short_name === "Toronto"
  );
};
