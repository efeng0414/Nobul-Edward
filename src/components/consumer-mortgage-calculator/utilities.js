export const amortizationRange = [5, 10, 15, 20, 25, 30, 35];
