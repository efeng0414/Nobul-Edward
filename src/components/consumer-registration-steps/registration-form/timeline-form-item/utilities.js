export const optionArray = [
  { value: 1, key: "timeline1" },
  { value: 2, key: "timeline2" },
  { value: 3, key: "timeline3" },
  { value: 4, key: "timeline4" },
  { value: 5, key: "timeline5" }
];
