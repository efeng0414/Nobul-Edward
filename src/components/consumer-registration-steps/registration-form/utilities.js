import { BUY, SELL, BOTH } from "../../../../core/constants/shared";

export const fieldMap = {
  LOOKING_FOR: "lookingFor",
  TIMELINE: "timeline",
  PRE_APPROVED_MORTGAGE: "preApprovedMortgage",
  AREA_OF_INTEREST: "areaOfInterest",
  TERM: "term",
  CONSENT: "consent",
  MORTGAGE_AMOUNT: "mortgageAmount",
  CONTRACT_TO_BROKERAGE: "contractToBrokerage",
  CONSENT_FOR_NEWS: "consent_for_news",
  CONSENT_FOR_EMAIL: "email_notifications"
};

export const getFormPayload = ({ signUpValues, formValues }) => {
  const postData = {
    ...signUpValues,
    ...formValues
  };
  postData.mortgageAmount = !postData.preApprovedMortgage
    ? "0"
    : postData.mortgageAmount.replace(/^0+/, "");

  postData.consentForNews = postData[fieldMap.CONSENT].includes(
    fieldMap.CONSENT_FOR_NEWS
  );

  postData.allowEmailNotifications = postData[fieldMap.CONSENT].includes(
    fieldMap.CONSENT_FOR_EMAIL
  );

  postData[fieldMap.LOOKING_FOR] = postData[fieldMap.LOOKING_FOR].includes(BOTH)
    ? [BUY, SELL]
    : [postData[fieldMap.LOOKING_FOR]];

  return postData;
};
