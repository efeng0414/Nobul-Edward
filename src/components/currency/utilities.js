export const CURRENCY_SYMBOLS = {
  CA: "<span class='currency__code'>CA</span>$",
  US: "<span class='currency__code'>US</span>$",
  EU: "€",
  GB: "£"
};
