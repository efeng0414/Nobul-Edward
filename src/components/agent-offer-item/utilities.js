import React from "react";
import { Icon, Tag } from "antd";
import { JOB_ACCEPTED, JOB_DELETED } from "../../../core/constants/jobs";
import {
  OFFER_OPEN,
  OFFER_ACCEPTED,
  OFFER_REJECTED,
  OFFER_WITHDRAWN,
  OFFER_PENDING_VERIFICATION
} from "../../../core/constants/offers";
import { translate } from "../../utilities/locale";

export const TAG_MAP = {
  ACCEPTED: {
    condition: ({ correspondingJob, offerDetail, withdraw }) =>
      !withdraw &&
      correspondingJob.status === JOB_ACCEPTED &&
      offerDetail.status === OFFER_ACCEPTED,
    jsx: ({ intl }) => (
      <Icon type="check" key="ACCEPTED">
        {translate(intl, "offerAccepted")}
      </Icon>
    )
  },
  OPEN: {
    condition: ({ correspondingJob, offerDetail, withdraw }) =>
      !withdraw && offerDetail.status === OFFER_OPEN,
    jsx: ({ intl }) => (
      <Tag className="agent-offer__tag agent-offer__tag--open" key="OPEN">
        {translate(intl, "offerOpen")}
      </Tag>
    )
  },
  REJECTED: {
    condition: ({ correspondingJob, offerDetail, withdraw }) =>
      !withdraw && offerDetail.status === OFFER_REJECTED,
    jsx: ({ intl }) => (
      <Tag
        className="agent-offer__tag agent-offer__tag--rejected"
        key="REJECTED"
      >
        {translate(intl, "offerRejected")}
      </Tag>
    )
  },
  WITHDRAWN: {
    condition: ({ correspondingJob, offerDetail, withdraw }) =>
      withdraw || offerDetail.status === OFFER_WITHDRAWN,
    jsx: ({ intl }) => (
      <Tag
        className="agent-offer__tag agent-offer__tag--withdrawn"
        key="WITHDRAWN"
      >
        {translate(intl, "offerWithdrawn")}
      </Tag>
    )
  },

  JOB_DELETED: {
    condition: ({ correspondingJob, offerDetail, withdraw }) =>
      !withdraw && correspondingJob.status === JOB_DELETED,
    jsx: ({ intl }) => (
      <Tag
        className="agent-offer__tag agent-offer__tag--job-deleted"
        key="JOB_DELETED"
      >
        {translate(intl, "offer.jobDeletedTag")}
      </Tag>
    )
  },

  PENDING_VERIFICATION: {
    condition: ({ correspondingJob, offerDetail, withdraw }) =>
      !withdraw && offerDetail.status === OFFER_PENDING_VERIFICATION,
    jsx: ({ intl }) => (
      <Tag
        className="agent-offer__tag agent-offer__tag--pending-verification"
        key="OPEN"
      >
        {translate(intl, "offerPendingVerification")}
      </Tag>
    )
  }
};
