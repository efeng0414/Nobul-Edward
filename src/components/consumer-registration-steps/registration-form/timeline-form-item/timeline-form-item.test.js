import React from "react";
import TimelineFormItem from "./index";
import { optionArray } from "./utilities";

describe("TimelineFormItem", () => {
  const componentProps = {
    getFieldDecorator: fieldName => jsx => [fieldName, jsx],
    intl: global.getIntl(),
    fieldName: "timeline"
  };

  const componentJSX = <TimelineFormItem {...componentProps} />;

  describe("Component Snapshot", () => {
    const componentSnapshot = global.renderer.create(componentJSX).toJSON();

    it("should match stored snapshot", () => {
      expect(componentSnapshot).toMatchSnapshot();
    });
  });

  describe("Component Render", () => {
    const componentMount = global.Enzyme.shallow(componentJSX);

    it("should render the correct number of options", () => {
      const optionCount = componentMount.find("Option").length;

      expect(optionCount).toBe(optionArray.length);
    });
  });
});
