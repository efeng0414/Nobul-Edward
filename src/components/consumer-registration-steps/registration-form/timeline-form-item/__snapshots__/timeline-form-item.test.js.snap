// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineFormItem Component Snapshot should match stored snapshot 1`] = `
<div
  className="consumer-registration-form-item"
>
  <div
    className="ant-row ant-form-item"
  >
    <div
      className="ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control"
      >
        <span
          className="ant-form-item-children"
        >
          <p
            className="consumer-registration-paragraph"
          >
            When in the near future are you looking buy or sell?
          </p>
          timeline
          <div
            className="consumer-registration-select ant-select ant-select-enabled"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          >
            <div
              aria-autocomplete="list"
              aria-expanded={false}
              aria-haspopup="true"
              className="ant-select-selection
            ant-select-selection--single"
              onKeyDown={[Function]}
              role="combobox"
              tabIndex={0}
            >
              <div
                className="ant-select-selection__rendered"
              />
              <span
                className="ant-select-arrow"
                onClick={[Function]}
                style={
                  Object {
                    "WebkitUserSelect": "none",
                    "userSelect": "none",
                  }
                }
                unselectable="on"
              >
                <b />
              </span>
            </div>
          </div>
        </span>
      </div>
    </div>
  </div>
</div>
`;
